### ISSUE TITLE
Add support to build from local source code

### ISSUE THREAD

**Issue by surajssd** [confidential-containers/cloud-api-adaptor#987]
Right now this [Azure Dockerfile](https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/azure/image/Dockerfile) based approach only supports the code to be pulled from somewhere. Also support building from the code locally available.

**Comment by bpradipt** [comment]
One of the reasons the Dockerfile didn't had support for local files was to keep the builds completely isolated and laying the foundation for builds in secure/isolated systems with SBOMs.

The alternative approach that was added was to replace the code with personal fork/branch via the CAA_SRC overrides.

May be we can explore using volume mounts for local sources.

**Comment by bpradipt** [comment]
> One of the reasons the Dockerfile didn't had support for local files was to keep the builds completely isolated and laying the foundation for builds in secure/isolated systems with SBOMs.
> 
> The alternative approach that was added was to replace the code with personal fork/branch via the CAA_SRC overrides.
> 
> May be we can explore using volume mounts for local sources.

@surajssd just wanted to share this aspect about local files.

**Comment by bpradipt** [comment]
> > @surajssd just wanted to share this aspect about local files.
> 
> I agree, I was wondering what can be the ideal method to do it. The current approach works from the CI perspective as well. Especially from the gh actions perspective, where we can use the checkout action and then build code inside the container.
> 

Yes, the reason for adding CAA_SRC_ was by @wainersm initially to handle the CI flow as well.  So imho let's keep the build isolated and not copy files from host. In your PR you may want to add a comment where CAA_SRC_ vars are there so that it's known to new developers getting onboarded.



> OTOH, the other complaints I have about the current approach is that we use `docker build` method to generate an artifact (pod vm image) that will be stored elsewhere (on Azure). And we don't use the container-image generated from it. I was wondering if there was a way to create generic builder images and we use that image using `docker run`?

So let me rephrase the question here for my understand. You want to create the packer based cloud images by using `docker run` instead of `docker build`.   
If my understanding is correct, then indeed it makes sense for cloud images. The reason we have `docker build` is primarily for the qcow2 images as the images are stored in the container itself.

**Comment by surajssd** [comment]
@bpradipt for #994 what do you suggest I should change? Should I add back the CAA_SRC_ ? The thing is we need a way to provide the local `variables.pkrvars.hcl` file from outside. Either mount it or copy it. So without that we can't build the final image for Azure at least, not sure about the other providers.
